### ISSUE TITLE
New Messenger design, API rate limit

### ISSUE THREAD

**Issue by pcmester** [Prabesh01/fb-message-delete-automation#1]
A few days ago, Facebook changed messenger.com to the new design, with API changes. While with some Chrome extension that reverts back to the old design, the keys can still be grabbed and the old API works, there are some caveats:

The old API is... old, so it will be removed soon
There is a rate limit, which blocks the removal for about an hour after 1-2 minutes of running.

**Comment by Prabesh01** [comment]
Edit: This comment was made on 2020. The Script stooped working from 2021.

**TLDR**: This script still works fine if you provide all the 4 values (c_user, doc_id, xs, dtsg) and used without making any other changes in the script (changes like commenting some lines, as mentioned in the script startup).

@pcmester, Thanks for opening the issue.  For now, this method is still working. I will always try my best to make this script work. I will update the script once it completely stops working and I get some free time.

As usual, contributions are always welcome. Share if you have any idea that would help this repo. Or you can make some changes in the repo by making  pull requests.

And about the rate limit, you are right. But regardless the fact, I see no other way to delete all messages of a convo (expect the one I mentioned at last of the readme file). That's why although it has rate limit, this is the only way to make the work done.

But to my surprise, I faced no rate limit when the script is used without making any changes. But we face rate limit if we commented the mentioned lines for customization as told by the script.  I, myself am not sure about how that works but, it does.
So, you will face rate limit if you tried out point number 2,3 or 4 as displayed on the script startup. Ergo, simply use the script as it is, just by keeping the required 4 values, and you will face no rate limit.

If you are facing problems like messages not being deleted after some use, try editing line 67 in the script and change it into: ```for count in range(total): ```. Just removed reversed() from the line, so it will start from the oldest messages. This too has worked when I last used the script.

**Comment by ucalyptus2** [comment]
@Prabesh01 any idea when the code will be updated ?

**Comment by Prabesh01** [comment]
I don't think it will be any soon. They have made vast changes which makes it too hard. Plus, since lock down is over, I wont have much free time. Everytime I come with sth, they patch it within few weeks so it seems worthless which make me less interested to work on it.

I would suggest to look for sth else or follow alternative methods as mentioned at the end of readme file. 

BTW, I liked your name @forkbabu :D
